# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderLayerManager.h
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 gfx/layers/wr/WebRenderLayerManager.h
--- a/gfx/layers/wr/WebRenderLayerManager.h	Fri Mar 22 18:28:31 2019 +0000
+++ b/gfx/layers/wr/WebRenderLayerManager.h	Fri Mar 22 18:28:42 2019 +0000
@@ -173,7 +173,9 @@
   WebRenderUserDataRefTable* GetWebRenderUserDataTable() {
     return mWebRenderCommandBuilder.GetWebRenderUserDataTable();
   }
-  WebRenderScrollData& GetScrollData() { return mScrollData; }
+  WebRenderScrollData& GetScrollData(wr::RenderRoot aRenderRoot) {
+    return mScrollDatas[aRenderRoot];
+  }
 
   void WrUpdated();
   void WindowOverlayChanged() { mWindowOverlayChanged = true; }
@@ -185,7 +187,10 @@
   void StopFrameTimeRecording(uint32_t aStartIndex,
                               nsTArray<float>& aFrameIntervals) override;
 
-  RenderRootStateManager* GetRenderRootStateManager() { return &mStateManager; }
+  RenderRootStateManager* GetRenderRootStateManager(
+      wr::RenderRoot aRenderRoot) {
+    return &mStateManagers[aRenderRoot];
+  }
 
  private:
   /**
@@ -206,7 +211,7 @@
 
   // This holds the scroll data that we need to send to the compositor for
   // APZ to do it's job
-  WebRenderScrollData mScrollData;
+  wr::RenderRootArray<WebRenderScrollData> mScrollDatas;
 
   bool mWindowOverlayChanged;
   bool mNeedsComposite;
@@ -231,9 +236,8 @@
   nsCString mURL;
   WebRenderCommandBuilder mWebRenderCommandBuilder;
 
-  size_t mLastDisplayListSize;
-
-  RenderRootStateManager mStateManager;
+  wr::RenderRootArray<size_t> mLastDisplayListSizes;
+  wr::RenderRootArray<RenderRootStateManager> mStateManagers;
 };
 
 }  // namespace layers